test/backend/test.py

```diff
@@ -87,11 +87,22 @@ class DummyBackend(onnx.backend.base.Backend):
     @classmethod
     def prepare(cls, model, device='CPU', **kwargs):
         super(DummyBackend, cls).prepare(model, device, **kwargs)
+        name = model.graph.name
+        model_name = name+".onnx"
+        exec_name = name + ".so"
+        # Clean the temporary files in case
+        os.system("rm -f " + model_name)
```

Comment:
It'll be safer to use temporary directories in python: https://stackoverflow.com/a/55104228 . If there's a bug in code logic like this the consequence can be severe. So we might want to delegate things like this to existing python modules.

To aid readability, you can give it a readable prefix/suffix as per TempDirectory documentation (https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory).

For a more complete example, you can see python code in our debug utility:
https://github.com/onnx/onnx-mlir/blob/50fca37336c1f8e8590949fb916abda1f60fdb85/utils/debug.py#L64

Reply:
The temporary result, *.onnx or *.so, may be needed for debugging, if a test fails.  These files are not temporary files to CMakefile
To avoid explicit delete in python, I think that we can call the clean automatically before each check-onnx-mlir, as Alex suggested. 
Since the entanglement of unitest and cmake, it seems no perfect solution.

Reply:
Just want to add that leaving *.onnx or *.so files on the disk by default may not be a good idea, given that model tests require *.onnx files with size between 10-500MB. This means test artifacts will take up gigabytes of space by default.

Any chance we can have them cleaned up by default, and in the case of failure or user request (e.g., through environment variable), keep the files?